testing/scenario_app/tool/run_android_tests_smoke.sh

```diff
@@ -8,11 +8,30 @@
 # why `./testing/scenario_app/run_android_tests.sh` did or did not report
 # failures correctly.
 
-# Run this command and print out the exit code.
-../third_party/dart/tools/sdks/dart-sdk/bin/dart ./testing/scenario_app/bin/android_integration_tests.dart \
-  --adb="../third_party/android_tools/sdk/platform-tools/adb" \
-  --out-dir="../out/android_debug_unopt_arm64" \
-  --smoke-test="dev.flutter.scenarios.EngineLaunchE2ETest"
+ADB="../third_party/android_tools/sdk/platform-tools/adb"
+OUT_DIR="../out/android_debug_unopt_arm64"
+SMOKE_TEST="dev.flutter.scenarios.EngineLaunchE2ETest"
 
-echo "Exit code: $?"
-echo "Done"
+# Optionally skip installation if -s is passed.
+if [ "$1" != "-s" ]; then
+  # Install the app and test APKs.
+  echo "Installing app and test APKs..."
+  $ADB install -r $OUT_DIR/scenario_app/app/outputs/apk/debug/app-debug.apk
+  $ADB install -r $OUT_DIR/scenario_app/app/outputs/apk/androidTest/debug/app-debug-androidTest.apk
+fi
+
+# Configure the device for testing.
+echo "Configuring device for testing..."
+$ADB shell settings put secure immersive_mode_confirmations confirmed
+
+# Reverse port 3000 to the device.
+echo "Reversing port 3000 to the device..."
+$ADB reverse tcp:3000 tcp:3000
```

Comment:
Where are these coming from?  Is this pulling logic from the android_integration_tests.dart, should there be a removal in that script?